src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/RazorFormattingService.cs

```diff
@@ -240,4 +241,48 @@ private static void UnwrapCSharpSnippets(TextEdit[] snippetEdits)
             snippetEdit.NewText = unwrappedText;
         }
     }
+
+    // This method counts the occurrences of CRLF and LF line endings in the original text. If LF line endings are more
+    // prevalent, it removes any CR characters from the text edits to ensure consistency with the LF style.
+    // This can be removed once we figure out how to get proper line ending information from the client.
+    private TextEdit[] NormalizeLineEndings(SourceText originalText, TextEdit[] minimalEdits)
+    {
+        var (crlfCount, lfCount) = CountLineEndings(originalText);
+        if (lfCount > crlfCount)
+        {
+            minimalEdits = minimalEdits
+                .Select(edit =>
+                {
+                    edit.NewText = edit.NewText.Replace("\r", string.Empty);
+                    return edit;
+                })
+                .ToArray();
```

Comment:
Given this is modifying the `edit` in place, there is no need to create a new array, so it would be a lot better to rewrite this as a simple foreach loop. ie

```
foreach (var edit in minimalEdits)
{
	edit.NewText = edit.NewText.Replace("\r", "");
}
```

Reply:
Good catch!